Assets/HSVPicker/UI/ColorPicker.cs

```diff
@@ -39,10 +39,27 @@ public Color CurrentColor
 
         private void OnEnable()
         {
-            SendChangedEvent();
+            if (Setup.RegenerateOnOpen)
+            {
+                Regenerate();
+            }
+            // no need to call this if Regenerate() has already been called.
+            else
+            {
+                SendChangedEvent();
```

Comment:
Go ahead and remove this, it causes issue #47. Also if you can, remove `SendChangedEvent()` on line 74 from `Regenerate()`.

Reply:
No problem, I've made those changes in the next commit